spec.emu

```diff
@@ -12,162 +12,113 @@ copyright: false
-    </emu-clause>
-    <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-step">
-        <h1>%NumericRangeIteratorPrototype%.step</h1>
-        <p>%NumericRangeIteratorPrototype%.step is an accessor property with attributes { [[Enumerable]]: false, [[Configurable]]: *true*}. The [[Get]] attribute is defined as follows:</p>
-        <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-step-getter">
-            <h1>get %NumericRangeIteratorPrototype%.step</h1>
-            <emu-alg>
-                1. Perform ? RequireInternalSlot(*this*, [[step]]).
-                1. Return *this*.[[step]].
-            </emu-alg>
+
+        <emu-clause id="sec-numeric-range-iterator-object">
+            <h1>The <dfn>NumericRangeIterator</dfn> Object</h1>
+            <p>A NumericRangeIterator object is an iterator that yields numbers. There is not a named constructor for NumericRangeIterator objects. Instead, NumericRangeIterator objects are created by the CreateNumericRangeIterator abstract operation as needed.</p>
+
+            <emu-clause id="sec-create-numeric-range-iterator" type="abstract operation">
+                <h1>
+                    CreateNumericRangeIterator (
+                        _start_: a Number or a BigInt,
+                        _end_: an ECMAScript language value,
+                        _option_: an ECMAScript language value,
+                        _type_: ~number-range~ or ~bigint-range~
+                    )
+                </h1>
+                <dl class="header"></dl>
+                <emu-alg>
+                    1. If _type_ is ~number-range~, let _primitiveType_ be *"number"*, else let _primitiveType_ be *"bigint"*.
+                    1. Assert: Type(_start_) is _primitiveType_.
+                    1. If _type_ is ~bigint-range~, let _zero_ be *0n*<sub>ℤ</sub>, else let _zero_ be *0*<sub>ℤ</sub>.
+                    1. If _type_ is ~bigint-range~, let _one_ be *1n*<sub>ℤ</sub>, else let _one_ be *1*<sub>ℤ</sub>.
+                    1. If _end_ is not *+∞* or *-∞* and if Type(_end_) is not _primitiveType_, throw a *TypeError* exception.
+                    <emu-note type="editor">Allowing all kinds of number (number, bigint, decimals, ...) to range from a finite number to infinity.</emu-note>
+                    1. If _start_ is *+∞* or *-∞*, throw a *RangeError* exception.
+                    1. Let _ifIncrease_ be end > start.
+                    1. Let _inclusiveEnd_ be *false*.
+                    1. If _option_ is *undefined* or *null*, let _step_ be *undefined*.
+                    1. Else if Type(_option_) is *object*, then
+                        1. Let _step_ be ? Get(_option_, "step").
+                        1. Let _inclusiveEnd_ be ToBoolean(? Get(_option_, "inclusive")).
+                    1. Else if Type(_option_) is _primitiveType_, let _step_ be _option_.
+                    1. Else, throw a *TypeError* exception.
```

Comment:
Similarly, I’d expect the options to be coerced ToObject when it’s non-nullish, rather than throwing.

Reply:
Why?

I guess maybe there's a consistency argument (though I'm not sure with what?) but it seems like that would generally be unintentional.

Reply:
about coerce `option`, it's OK to me, but since I decided to not coerce `start`, I think it will be more consistent.

Reply:
Let me step back - what does Temporal do with options bags? We should at least be consistent with that.

Reply:
https://tc39.es/proposal-temporal/#sec-getoptionsobject

Reply:
alrighty, fair enough. let's inline those same AOs, then, and use them.

Reply:
Temporal does what ECMA-402 does (at least, what ECMA-402 does for new APIs; there are some older Intl APIs that coerce primitives to objects). See https://github.com/tc39/ecma402/issues/480